### ISSUE TITLE
qdrant_client-1.6.9: "Can't init WAL" when creating simple collection

### ISSUE THREAD

**Issue by mfirry** [qdrant/qdrant#3132]
I'm getting `error processing request: Can't init WAL: Os { code: 22, kind: InvalidInput, message: "Invalid argument" }` just by trying to create a new collection as suggested on the docs

I'm just doing `client.create_collection("2db5aa02", vectors_config)` where `vectors_config` is `models.VectorParams(size=100, distance=models.Distance.COSINE)`.

Using `qdrant_client-1.6.9` and using locally `docker.io/qdrant/qdrant:latest`

**Comment by bashofmann** [comment]
I get the same error with the unprivileged image:

```
podman run -p 6333:6333 -p 6334:6334 -v ~/tmp/qdrant_storage:/qdrant/storage --userns=keep-id:uid=1000,gid=1000 qdrant/qdrant:latest-unprivileged
```

I found this issue here https://github.com/containers/podman/issues/17560, so I tried the --userns argument.

As soon as I try to add data:

```
2023-12-06T11:37:28.884705Z  WARN qdrant::actix::helpers: error processing request: Can't init WAL: Os { code: 22, kind: InvalidInput, message: "Invalid argument" }
```

I don't think it's "only" a simple permission issue. The volume actually contains data:

```
qdrant@7fb40fb4ef7b:/qdrant/storage$ ls -la
total 4
drwxr-xr-x. 5 qdrant nogroup 160 Dec  6 11:37 .
drwxr-xr-x. 1 qdrant qdrant   65 Dec  6 11:37 ..
drwxr-xr-x. 3 qdrant nogroup  96 Dec  6 11:37 aliases
drwxr-xr-x. 3 qdrant nogroup  96 Dec  6 11:37 collections
-rw-r--r--. 1 qdrant nogroup 299 Dec  6 11:37 raft_state.json
```

So qdrant seems to be able to write to it.

**Comment by T1T4N** [comment]
I managed to solve the issue on macOS 14.3 (23D56) Sonoma M1 with the following command:
`podman run -p 6333:6333 -p 6334:6334 --mount type=bind,src=$PWD/qdrant_storage,target=/storage qdrant/qdrant`

podman version:
```
Client:       Podman Engine
Version:      4.9.2
API Version:  4.9.2
Go Version:   go1.21.6
Git Commit:   f9a48ebcfa9a39144be0f86f4ba842752835f945
Built:        Fri Feb  2 22:27:01 2024
OS/Arch:      darwin/arm64

Server:       Podman Engine
Version:      4.8.3
API Version:  4.8.3
Go Version:   go1.21.5
Built:        Wed Jan  3 15:10:40 2024
OS/Arch:      linux/arm64
```

**Comment by T1T4N** [comment]
I managed to fix this issue on macOS 14.3 (23D56) Sonoma M1 by running the following commands:

```
export CONTAINERS_MACHINE_PROVIDER=applehv
podman machine init --rootful
podman machine start
podman run -p 6333:6333 -p 6334:6334 --privileged -v $PWD/qdrant_storage:/qdrant/storage qdrant/qdrant
```

**Comment by mfirry** [comment]
thanks @T1T4N but now i get the following error when i try and create a collection
```
2024-02-07T12:36:15.907068Z  INFO storage::content_manager::toc::collection_meta_ops: Creating collection 4e55ff00
2024-02-07T12:36:15.921970Z  WARN qdrant::actix::helpers: error processing request: File IO error: Permission denied (os error 13)
```

not sure this is related

**Comment by timvisee** [comment]
> 2024-02-07T12:36:15.921970Z  WARN qdrant::actix::helpers: error processing request: File IO error: Permission denied (os error 13)

According to this your permissions still are not set up correctly. Could you maybe verify this manually, whether you're able to create a file on that volume?

**Comment by mfirry** [comment]
I agree. How should I check though? On the host machine (my laptop), my user is definitely able to create files there yes.

**Comment by timvisee** [comment]
> I agree. How should I check though? On the host machine (my laptop), my user is definitely able to create files there yes.

Run a command within the container itself to test this.

First start your container, and then:

```bash
$ docker run -it qdrant bash
$ cd /qdrant/storage
$ touch test
$ ls
```

This should create a file called `test`, which you should see when running `ls`.

Does that work?

**Comment by mfirry** [comment]
Ok. thanks @timvisee for the help. 

the solution suggested by @T1T4N actually makes it work. not sure why these things are needed to be honest (`applehv`, `rootful` and `privileged`) but it does work now. thanks

**Comment by T1T4N** [comment]
@mfirry Happy to help! This has been an interesting learning experience for me as well.
For the record, `rootful` shouldn't be necessary, I just double-checked locally and just `applehv + privileged` works.
I assume it has something to do with the way user namespaces are implemented on macOS (or the lack thereof).

**Comment by T1T4N** [comment]
Following the [Podman troubleshooting guide](https://github.com/containers/podman/blob/main/troubleshooting.md#2-cant-use-volume-mount-get-permission-denied), specifying `--security-opt label=disable` instead of `--privileged` also works!
Tested with `applehv`
